mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -719,8 +719,12 @@ impl Nexus {
 
     /// read vectored io from the underlying children.
     pub(crate) fn readv(&self, io: &Bio, channels: &mut NexusChannelInner) {
-        // we use RR to read from the children also, set that we only need
-        // to read from one child before we complete the IO to the callee.
+        // we use RR to read from the children.
+        if channels.read_ch.len() == 0 {
```

Comment:
don't think this can ever happen, should it be an assert?

Reply:
Ok maybe an assert will do. But we do clear the vectors when we reconfigure and I wouldn't bet there couldn't be a window when an IO gets processed at that moment.

Reply:
that's a good point in favour of keeping this one in then though I guess we shouldn't be processing IO's while the channels are being reconfigured otherwise we'd open the doors to data corruption?

Reply:
Actually I think you are right. refresh_io_channels() happens in the same thread that the IO belongs to so refresh() is effectively atomic as far as IO is concerned. I'll make it an assert.